### ISSUE TITLE
Retrieve grid data on re-draw/render

### ISSUE THREAD

**Issue by joshwpfx** [TonyGermaneri/canvas-datagrid#249]
Hi Tony!
I've been reviewing the documentation, and I can't seem to figure out how to grab the grid data after a particular user filter. Is there something similar to https://datatables.net/reference/event/draw available with this?

Scenario if helpful: When a user filters and changes the grid data, this data must then push to a chart.js graph.

I see I can add a custom contextmenu, but I'd like to stick with the out-of-the-box filtering! Thanks, please let me know if you have any questions or feedback.

**Comment by TonyGermaneri** [comment]
I have a ticket for this improvement #241.  Seems there's a few people waiting on it.  Sorry about that.  I'll put some work into this weekend.  For now you should filter externally and apply the data to the grid.  If you need to use the grids filters and capture the filtered data, I think you can just access it with `grid.data`, if it is unfiltered, you can pass it through the filter list `grid.filter`.  But I'm fairly sure the data is filtered.   After ticket #241 is done, it will be a bit more clear how to get filtered vs. unfiltered data. 

I really hope this is even the question you are asking.  Let me know if I made things more clear, or perhaps more confusing.

**Comment by joshwpfx** [comment]
I 'think' this is what I was looking for, but I'm not 100% certain.
If pseudo-code helps:
- data set on grid
- user filters via built-in select
- **at this point it would be nice to have the new filtered data-set in an easier event, perhaps a draw/build event similar to datatables js**

So you're saying for the time-being, I should:
- remove the current context menus related to filtering
- use external selects of my data
- on change, make these update the data-set of the grid

Seems a bit tedious, but it sounds like that's my easiest option at the moment?

Thanks so much!

**Comment by TonyGermaneri** [comment]
Sorry, I completely misunderstood your question.

`grid.data` will hold the filtered data.

The grid does fire a number of events during the draw cycle, several per cell, but likely they fire too frequently to be of real use.  

It may seem strange, but the `resize` event is probably what you want to listen for.  Resize will be fired after the filter is set, that will give you a chance to see the filtered data `grid.data`.

I hope that helps.  Sorry for the confused response before.

**Comment by marionicolas** [comment]
I know its been a long time... and the project seems to be a little abandoned, but... didnt get it, is there a reasonable way to get the actual filtered/ordered data? because grid.data gives just the original data.
